src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs

```diff
@@ -193,7 +193,7 @@ private static CommandLineApplication InitializeApp(string[] args, CommandOutput
                 app.Name = DotnetPackageAppName;
                 AddPackageReferenceCommand.Register(app, () => log, () => new AddPackageReferenceCommandRunner());
                 RemovePackageReferenceCommand.Register(app, () => log, () => new RemovePackageReferenceCommandRunner());
-                ListPackageCommand.Register(app, () => log, () => new ListPackageCommandRunner());
+                ListPackageCommand.Register(app, () => loggerFunc(log)(), () => new ListPackageCommandRunner());
             }
             else
             {
```

Comment:
By the way, `loggerFunc` below 👇 is invoked outside of the delegate, so `DeleteCommand`, `PushCommand`, and `LocalsCommands` all have `HidePrefixForInfoAndMinimal = true`. From the shape of the code it doesn't look intentional but 🤷.

Reply:
Are you sure? `Register` takes a Func<ILogger>, didn't think that would be an invocation.

Reply:
On the previous line `CommandParsers.Register(app, loggerFunc(log));`, the `loggerFunc` is being invoked thus changing the state of `HidePrefixForInfoAndMinimal` but also returning a function.